clash-lib/src/Clash/Netlist/BlackBox/Util.hs

```diff
@@ -315,6 +317,39 @@ setSym bbCtx l = do
 
 type FileName = FilePath
 
+-- | Canonicalizes a datafile path, keeping in mind that the path might be
+-- relative to one of the import directories passed in with @-i@.
+canonicalizeDataFilePath
+  :: [FilePath]
+  -- ^ Import directories passed in with @-i@
+  -> FilePath
+  -- ^ File to canonicalize
+  -> IO FilePath
+canonicalizeDataFilePath idirs toCanonicalize
```

Comment:
We're (veeeeery slowly) converting to "new style" type signatures. I.e.,

```haskell
daFunction ::
  a ->
  b -> 
  c
```

could you do that here too?